spark/src/main/java/org/apache/iceberg/spark/SparkSchemaUtil.java

```diff
@@ -63,6 +65,21 @@ public static Schema schemaForTable(SparkSession spark, String name) {
     return new Schema(converted.asNestedType().asStructType().fields());
   }
 
+  /**
+   * Given a Spark table identifier, determine the PartitionSpec.
+   * @param spark the SparkSession which contains the identifier
+   * @param table a TableIdentifier, if the namespace is left blank the catalog().currentDatabase() will be used
+   * @return a IcebergPartitionSpec representing the partitioning of the Spark table
+   * @throws AnalysisException if thrown by the Spark catalog
+   */
+  public static PartitionSpec specForTable(SparkSession spark, TableIdentifier table) throws AnalysisException {
+    String db = table.database().nonEmpty() ? table.database().get() : spark.catalog().currentDatabase();
+    PartitionSpec spec = identitySpec(
+        schemaForTable(spark, table.unquotedString()),
```

Comment:
I think that you can get the table DDL via a utility function in catalyst, and then there's a function that returns a schema from DDL. However, if this is returning an Iceberg schema than do please ignore me :)

Reply:
Yeah this is the IcebergSchema :)

Reply:
I figured once I typed it out. I should really start deleting my comments once I come to the answer rubber duck debugging with myself in the github comments 😅 .